File: src/main/java/net/ccbluex/liquidbounce/render/ultralight/hooks/UltralightIntegrationHook.java
package net.ccbluex.liquidbounce.render.ultralight.hooks;

import de.vincentschweiger.phantomclient.events.EventAnnotation;
import de.vincentschweiger.phantomclient.events.impl.*;
import net.ccbluex.liquidbounce.render.ultralight.RenderLayer;
import net.ccbluex.liquidbounce.render.ultralight.UltralightEngine;

public class UltralightIntegrationHook {

    @EventAnnotation
    public void onKeyboardKey(KeyboardKeyEvent it) {
        UltralightEngine.inputAdapter.keyCallback(it.getWindow(), it.getKey(), it.getScancode(), it.getI(), it.getJ());
    }

    @EventAnnotation
    public void onKeyboardChar(KeyboardCharEvent event) {
        UltralightEngine.inputAdapter.charCallback(event.getWindow(), event.getI());
    }

    @EventAnnotation
    public void onMouseScrool(MouseScrollEvent event) {
        UltralightEngine.inputAdapter.scrollCallback(event.getWindow(), event.getHorizontal(), event.getVertical());
    }

    @EventAnnotation
    public void onMouseCursor(MouseCursorEvent event) {
        UltralightEngine.inputAdapter.cursorPosCallback(event.getWindow(), event.getX(), event.getY());
    }

    @EventAnnotation
    public void onRenderGameOverlay(RenderOverlayEvent event) {
        UltralightEngine.INSTANCE.render(RenderLayer.OVERLAY_LAYER, event.getMatrixStack());
    }

    @EventAnnotation
    public void onRenderScreen(ScreenRenderEvent event) {
        UltralightEngine.INSTANCE.render(RenderLayer.SCREEN_LAYER, event.getMatrixStack());
    }

    @EventAnnotation
    public void onGameRender(GameRenderEvent event) {
        UltralightEngine.INSTANCE.update();
    }

    @EventAnnotation
    public void onWindowResize(WindowResizeEvent event) {
        UltralightEngine.INSTANCE.resize(event.getWidth(), event.getHeight());
    }

    @EventAnnotation
    public void onWindowFocus(WindowFocusEvent event) {
        UltralightEngine.inputAdapter.focusCallback(event.getWindow(), event.isFocused());
    }

    @EventAnnotation
    public void onMouseButton(MouseButtonEvent event) {
        UltralightEngine.inputAdapter.mouseButtonCallback(event.getWindow(), event.getButton(), event.getAction(), event.getMods());
    }
}

File: src/main/kotlin/me/cookie/abyssclient/render/ultralight/hooks/UltralightIntegrationHook.kt
package me.cookie.abyssclient.render.ultralight.hooks


import me.cookie.abyssclient.event.*
import me.cookie.abyssclient.render.ultralight.RenderLayer
import me.cookie.abyssclient.render.ultralight.UltralightEngine

/**
 * A integration bridge between Minecraft and Ultralight
 */
object UltralightIntegrationHook : Listenable {

    val gameRenderHandlerHandler = handler<GameRenderEvent> {
        UltralightEngine.update()
    }

    val screenRenderHandler = handler<ScreenRenderEvent> {
        UltralightEngine.render(RenderLayer.SCREEN_LAYER, it.matrices)
    }

    val overlayRenderHandler = handler<OverlayRenderEvent> {
        UltralightEngine.render(RenderLayer.OVERLAY_LAYER, it.matrices)
    }

    val windowResizeWHandler = handler<WindowResizeEvent> {
        UltralightEngine.resize(it.width.toLong(), it.height.toLong())
    }

    val windowFocusHandler = handler<WindowFocusEvent> {
        UltralightEngine.inputAdapter.focusCallback(it.window, it.focused)
    }

    val mouseButtonHandler = handler<MouseButtonEvent> {
        UltralightEngine.inputAdapter.mouseButtonCallback(it.window, it.button, it.action, it.mods)
    }

    val mouseScrollHandler = handler<MouseScrollEvent> {
        UltralightEngine.inputAdapter.scrollCallback(it.window, it.horizontal, it.vertical)
    }

    val mouseCursorHandler = handler<MouseCursorEvent> {
        UltralightEngine.inputAdapter.cursorPosCallback(it.window, it.x, it.y)
    }

    val keyboardKeyHandler = handler<KeyboardKeyEvent> {
        UltralightEngine.inputAdapter.keyCallback(it.window, it.keyCode, it.scancode, it.action, it.mods)
    }

    val keyboardCharHandler = handler<KeyboardCharEvent> {
        UltralightEngine.inputAdapter.charCallback(it.window, it.codepoint)
    }

}